### PR TITLE
Extend dual-stack documentation concerning DNS considerations

### PR DESCRIPTION
**Pull request by ScheererJ**
**How to categorize this PR?**

/area networking
/area documentation
/kind enhancement
/platform aws

**What this PR does / why we need it**:

Extend dual-stack documentation concerning DNS considerations.

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```other operator

```

/invite @axel7born @MartinWeindel

### CODE REVIEW

File: docs/usage/dual-stack-ingress.md [hunk 1]
```diff
@@ -91,3 +91,26 @@ connections it may lead to wasted/forgotten resources. Therefore, the (manual) c
...
+
+If there are DNS entries directly linked to the corresponding load balancer and they are managed by the
+[shoot-dns-service](https://github.com/gardener/gardener-extension-shoot-dns-service), you can identify this via
+annotations with the prefix `dns.gardener.cloud/`. Those annotations can be linked to a `Service`, `Ingress` or
+`Gateway` resources. Alternatively, they may also use `DNSEntry` or `DNSAnnotation` resources.
+
+To prevent direct updates of the DNS entries when the load balancer is migrated add the annotation
+`dns.gardener.cloud/ignore: 'true'` to all affected resources next to the other `dns.gardener.cloud/...` annotations
+before starting the migration. Next, migrate the load balancer to be dual-stack enabled. Once the load balancer has
+been provisioned, i.e. after you can resolve its technical domain name available in the `Service` under
```
Comment (Sven1410) [inline docs/usage/dual-stack-ingress.md:110]: the technical name in "status:" is updated immediately - this cannot be used as an indicator that the loadbalancer is ready.

Comment (ScheererJ) [reply docs/usage/dual-stack-ingress.md:110]: Why not? You can use the name and try to resolve it via `nslookup`/`dig`. Until there are IP addresses it is certainly not ready. Furthermore, you can try to connect to this name depending on your workload.
Should I rephrase this to make it more clear? What did you have in mind or how would you test it?

Comment (Sven1410) [reply docs/usage/dual-stack-ingress.md:110]: I only checked the status in aws.

_"after you can resolve its technical domain name available in the `Service` under `status.loadBalancer.ingress` "_ is confusing a bit, because you can do that immediately. But you are correct: _"and you can successfully connect to your service via this address"_  that is the important point. I think I have overseen this second part of the sentence. But additionally I have no idea how to use this name (e.g. "k8s-istioing-gateway-6a1b8c8789-aeb5e2303ea96d20.elb.us-east-1.amazonaws.com") to test my service. None of our services will respond on this URL, it delivers ERR_CONNECTION_TIMED_OUT.

Comment (Sven1410) [reply docs/usage/dual-stack-ingress.md:110]: it works for ingress-nginx , e.g.: 
http://k8s-ingressn-ingressn-09c392f371-cc6ee1d2716ffb1b.elb.eu-central-1.amazonaws.com/
it delivers at least _"default backend - 404"_
but the istio gateway is not responding on its aws url (ERR_CONNECTION_TIMED_OUT)

Comment (Sven1410) [reply docs/usage/dual-stack-ingress.md:110]: but ok - you wrote in the comment _"try to resolve it via nslookup/dig"_ that is maybe a good method (but an untested method :-) maybe aws has the same problem and propagates the dns name to early)

Comment (ScheererJ) [reply docs/usage/dual-stack-ingress.md]: I rephrased the section to make it more clear that there are multiple options. Frankly, depending on the service to be migrated some options might be easier than others and waiting some time is also a valid option.
WDYT?

Comment (Sven1410) [reply docs/usage/dual-stack-ingress.md]: should be clear now - much better.

File: docs/usage/dual-stack-ingress.md [hunk 2]
```diff
@@ -91,3 +91,50 @@ connections it may lead to wasted/forgotten resources. Therefore, the (manual) c
 when migrating an existing `Service` instance.
 
 For more details see [AWS Load Balancer Documentation - Network Load Balancer](https://kubernetes-sigs.github.io/aws-load-balancer-controller/v2.4/guide/service/nlb/).
+
+## DNS Considerations to prevent Downtime during a dual-stack Migration
```
Comment (MartinWeindel) [inline docs/usage/dual-stack-ingress.md:95]: Consistent capatilization
```suggestion
## DNS Considerations to Prevent Downtime During a Dual-Stack Migration
```

File: docs/usage/dual-stack-ingress.md [hunk 3]
```diff
@@ -91,3 +91,50 @@ connections it may lead to wasted/forgotten resources. Therefore, the (manual) c
...
+
+In case the migration of an existing service is desired, please check if there are DNS entries directly linked to the
+corresponding load balancer. The migrated load balancer will directly have a new domain name, which will not be ready
```
Comment (MartinWeindel) [inline docs/usage/dual-stack-ingress.md:98]:
```suggestion
corresponding load balancer. The migrated load balancer will have a new domain name immediately, which will not be ready
```

File: docs/usage/dual-stack-ingress.md [hunk 4]
```diff
@@ -91,3 +91,50 @@ connections it may lead to wasted/forgotten resources. Therefore, the (manual) c
...
+
+-  Check in the AWS console for the corresponding load balancer provisioning state
+-  Perform domain name lookups with `nslookup`/`dig` to check whether the name resolves to an IP address.
+-  Call your workload via the new load balancer, e.g. using
+  `curl --resolve <my-domain-name>:<port>:<IP-address> https://<my-domain-name>:<port>`, which allows you to call your
+  service with the "correct" domain name without using actual name resolution.
+- Wait a fixed period of time as load balancer creation is usually finished within 15 minutes
+
+Once the load balancer has been provisioned, you can remove the annotation `dns.gardener.cloud/ignore: 'true'` again
+from the affected resources. It may take some additional time until the domain name change finally propagates.
```
Comment (MartinWeindel) [inline docs/usage/dual-stack-ingress.md:140]:
```suggestion
from the affected resources. It may take some additional time until the domain name change finally propagates (up to one hour).
```

**Review by MartinWeindel — COMMENTED** [review]
lgtm, only some minor suggestions

**Review by AndreasBurger — APPROVED** [review]
